streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -502,6 +517,13 @@ public StreamsConfig(final Map<?, ?> props) {
         consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, originals().get(BOOTSTRAP_SERVERS_CONFIG));
         // remove deprecate ZK config
         consumerProps.remove(ZOOKEEPER_CONNECT_CONFIG);
+        if (eosEnabled) {
+            if (clientProvidedProps.containsKey(ConsumerConfig.ISOLATION_LEVEL_CONFIG)) {
```

Comment:
The point is, that the user cannot set this config, and there is only _one_ valid value. If there would be multiple valid values, it would make sense to throw (as we should not "randomly" pick one) -- but as there is only one, we can just set it an run (and log). This gives a better out-of-the-box experience IMHO -- it's more "resilient" and "self-healing" if you wish (IMHO, less exceptions are always better if the exceptions are not really needed as there is only one fix)